### DISCUSSION TITLE
Solve the convergence problem

### DISCUSSION THREAD

**Discussion by amosaha** [idaholab/moose#19407] [Q&A Modules: Solid mechanics] [answered]
hello,
I made a thermal expansion model, but its results have not converged. I have tried to modify it many times, but I still can't get the result. I wonder if I have missed some important conditions?

[themal expansion.txt](https://github.com/idaholab/moose/files/7544489/themal.expansion.txt)

The general content of my model is: block2 is the heat source, and the thermal expansion of the material needs to be observed.
The code and graphics are attached here.
The pink area is block2.
![image](https://user-images.githubusercontent.com/93416273/141946284-9ea2b408-3e1b-46f8-9fac-c940902a9c70.png)
![image](https://user-images.githubusercontent.com/93416273/141946742-c23f4eba-ea89-4eb6-aa80-4003d3c487cc.png)

![image](https://user-images.githubusercontent.com/93416273/141946428-9e338336-6795-4069-8445-79061dd1add2.png)

**Comment by sapitts** [comment, marked as answer]
Hi @amosaha,  
You will need to add some displacement boundary conditions to your input file, along with the temperature DirchletBCs you currently have included. I'd recommend taking a look at the [TensorMechanics Introductory Tutorial)[https://mooseframework.inl.gov/modules/tensor_mechanics/tutorials/introduction/index.html] and the documentation page for [ComputeThermalExpansionEigenstrain](https://mooseframework.inl.gov/source/materials/ComputeThermalExpansionEigenstrain.html) for examples of how to set up this type of problem.

Hope this helps,
Stephanie

**Reply by GiudGiud** [reply]
See https://github.com/idaholab/moose/discussions/19314, we discussed this for a cylindrical case

**Reply by amosaha** [reply]
I have added  the displacement boundary conditions to my my input file,but this problem has not been solved.Do I need to add other parts if I want to couple thermal diffusion and thermal expansion? My input file does not have this description.

**Reply by GiudGiud** [reply]
I dont see them in your input. 

Did you manage to get convergence for a thermal diffusion only problem first?
